WalletWasabi.Fluent/State/StateMachine.cs

```diff
@@ -51,16 +47,25 @@ public void Fire(TTrigger trigger)
 			Goto(trigger, _currentState.Parent.StateId, true, false);
 			Goto(trigger, _currentState.GetDestination(trigger));
 		}
+		else
+		{
+			_currentState.Process(trigger);
+		}
 	}
 
 	public void Start()
 	{
 		_currentState.Enter();
 	}
 
-	public void Process()
+	private void RegisterState(TState state)
 	{
-		_currentState.Process();
+		if (!_states.ContainsKey(state))
```

Comment:
In case the state is not found in the `_states` the it means that something is not okay somewhere and that situation shouldn't be ignored. The pattern where a `void` method's body consists only of an `if` statement without else is suspicious.